### PR TITLE
refactor: BackTop cssinjs 

### PR DESCRIPTION
**Pull request by LongHaoo**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      BackTop cssinJs change     |
| 🇨🇳 Chinese |    改造BackTop CssInJS       |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by LongHaoo** [comment]
先提个pr还没搞完

File: components/back-top/style/index.tsx [hunk 1]
```diff
@@ -1,2 +1,140 @@
...
+          // FIX ME @back-top-hover-bg
+          backgroundColor: token.backTopHoverBackground,
+          transition: 'all 0.3s',
+        },
+      },
+
+      // change to .backtop .backtop-icon
+      [`${backTopCls}-icon`]: {
+        // FIXME
+        fontSize: '24px',
```
Comment (zombieJ) [inline components/back-top/style/index.tsx:76]: `24`

File: components/back-top/style/index.tsx [hunk 2]
```diff
@@ -1,2 +1,140 @@
...
+        // FIXME @back-top-bg
+        backgroundColor: token.backTopBackground,
+        // FIXME
+        borderRadius: '20px',
+        transition: 'all 0.3s',
+
+        '&:hover': {
+          // FIX ME @back-top-hover-bg
+          backgroundColor: token.backTopHoverBackground,
+          transition: 'all 0.3s',
```
Comment (zombieJ) [inline components/back-top/style/index.tsx:69]: `token.durationSlow`

File: components/back-top/style/index.tsx [hunk 3]
```diff
@@ -1,2 +1,140 @@
...
+        height: '40px',
+        overflow: 'hidden',
+        // FIXME @back-top-color
+        color: token.backTopColor,
+        textAlign: 'center',
+        // FIXME @back-top-bg
+        backgroundColor: token.backTopBackground,
+        // FIXME
+        borderRadius: '20px',
+        transition: 'all 0.3s',
```
Comment (zombieJ) [inline components/back-top/style/index.tsx:64]: token.durationSlow

File: components/back-top/style/index.tsx [hunk 4]
```diff
@@ -1,2 +1,140 @@
...
+        width: '40px',
+        height: '40px',
+        overflow: 'hidden',
+        // FIXME @back-top-color
+        color: token.backTopColor,
+        textAlign: 'center',
+        // FIXME @back-top-bg
+        backgroundColor: token.backTopBackground,
+        // FIXME
+        borderRadius: '20px',
```
Comment (zombieJ) [inline components/back-top/style/index.tsx:63]: `20`

File: components/back-top/style/index.tsx [hunk 5]
```diff
@@ -1,2 +1,140 @@
...
+      zIndex: token.zIndexPopup,
+      width: '40px',
+      height: '40px',
+      cursor: 'pointer',
+
+      '&:empty': {
+        display: 'none',
+      },
+
+      '&-rtl': {
```
Comment (zombieJ) [inline components/back-top/style/index.tsx:46]: 使用逻辑位置，不要出现 left right xxxLeft xxxRight

File: components/back-top/style/index.tsx [hunk 6]
```diff
@@ -1,2 +1,140 @@
...
+const genSharedBackTopStyle: GenerateStyle<BackTopToken> = (token): CSSObject => {
+  const { backTopCls } = token;
+
+  return {
+    [backTopCls]: {
+      ...resetComponent(token),
+
+      position: 'fixed',
+      // FIXME
+      right: '100px',
```
Comment (zombieJ) [inline components/back-top/style/index.tsx:33]: 检查一下所有数字，除了 lineHeight 不要带上 px。

File: components/back-top/style/index.tsx (hunk 5 above)
Comment (zombieJ) [reply components/back-top/style/index.tsx:46]: 上下都检查一下~

File: components/back-top/style/index.tsx [hunk 7]
```diff
@@ -1,2 +1,136 @@
...
+const genSharedBackTopStyle: GenerateStyle<BackTopToken> = (token): CSSObject => {
+  const { backTopCls } = token;
+
+  return {
+    [backTopCls]: {
+      ...resetComponent(token),
+
+      position: 'fixed',
+      // FIXME
+      right: 100,
```
Comment (zombieJ) [inline components/back-top/style/index.tsx:35]: 要用逻辑位置代替

File: components/back-top/style/index.tsx [hunk 8]
```diff
@@ -1,2 +1,136 @@
...
+
+  return [
+    useStyleRegister({ theme, token, hashId, path: [prefixCls] }, () => {
+      const { BackTop } = token;
+
+      const backTopBackground = 'rgb(16, 136, 233)';
+      const backTopColor = '#fff';
+      const backTopHoverBackground = '#000000d9';
+
+      const durationSlow = 3;
```
Comment (zombieJ) [inline components/back-top/style/index.tsx:116]: token 里有 durationSlow，不用再定义一遍~

File: components/back-top/style/index.tsx [hunk 9]
```diff
@@ -1,2 +1,136 @@
...
+    [`@media (max-width: ${token.screenXS}px)`]: {
+      [backTopCls]: {
+        marginInlineEnd: 20,
+      },
+    },
+  };
+};
+
+// ============================== Export ==============================
+export default function useStyle(prefixCls: string): UseComponentStyleResult {
```
Comment (zombieJ) [inline components/back-top/style/index.tsx:105]: 参考一下这个，使用统一方法：https://github.com/ant-design/ant-design/blob/c86e8fa0a26828a6ebe5f7e2ce172d78706ff3fd/components/button/style/index.tsx#L373

File: components/back-top/style/index.tsx [hunk 10]
```diff
@@ -1,2 +1,136 @@
...
+      const backTopBackground = 'rgb(16, 136, 233)';
+      const backTopColor = '#fff';
+      const backTopHoverBackground = '#000000d9';
+
+      const durationSlow = 3;
+
+      const backTopToken: BackTopToken = {
+        ...token,
+
+        backTopCls: `.${prefixCls}`,
```
Comment (zombieJ) [inline components/back-top/style/index.tsx:121]: 使用统一方法后，这个 cls 会自动生成

File: components/back-top/style/index.tsx (hunk 8 above)
Comment (LongHaoo) [reply components/back-top/style/index.tsx:116]: token里面没有durationSlow

**Comment by zombieJ** [comment]
Rebase 一下~~

**Pull request merged**